Mage.Sets/src/mage/cards/o/OptimusPrimeAutobotLeader.java

```diff
@@ -0,0 +1,200 @@
+package mage.cards.o;
+
+import mage.MageInt;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.AttacksWithCreaturesTriggeredAbility;
+import mage.abilities.effects.common.TransformSourceEffect;
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.abilities.effects.keyword.BolsterEffect;
+import mage.abilities.keyword.LivingMetalAbility;
+import mage.abilities.keyword.TrampleAbility;
+import mage.abilities.keyword.TransformAbility;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.events.DamagedEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.targetpointer.FixedTargets;
+import mage.watchers.Watcher;
+
+import java.util.*;
+
+/**
+ * @author jbureau88
+ */
+public final class OptimusPrimeAutobotLeader extends CardImpl {
+    public OptimusPrimeAutobotLeader(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.VEHICLE);
+        this.power = new MageInt(6);
+        this.toughness = new MageInt(8);
+        this.color.setWhite(true);
+        this.color.setBlue(true);
+        this.color.setRed(true);
+        this.nightCard = true;
+
+        // Living metal
+        this.addAbility(new LivingMetalAbility());
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        this.getSpellAbility().addWatcher(new OptimusPrimeAutobotLeaderWatcher());
+
+        // Whenever you attack, bolster 2.
+        this.addAbility(new AttacksWithCreaturesTriggeredAbility(new BolsterEffect(2), 1));
+
+        // The chosen creature gains trample until end of turn.
+        this.addAbility(new AutobotLeaderAddTrampleTriggeredAbility());
```

Comment:
This is a hacky way to implement it and could trigger off unrelated counters being added. The watcher checking name is also brittle. I think what actually needs to happen is `BolsterEffect` gets an optional second parameter to take an additional `Effects` to add to the chosen creature. Then this card can add the gain trample effect, and the delayed triggered ability that stores a MageObjectReference to the source and converts it if possible.